core/cloudflow-operator/src/main/scala/cloudflow/operator/runner/FlinkRunner.scala

```diff
@@ -106,15 +106,17 @@ object FlinkRunner extends Runner[CR] {
     val name      = Name.ofFlinkApplication(deployment.name)
     val appLabels = CloudflowLabels(app)
     val labels = appLabels.withComponent(name, CloudflowLabels.StreamletComponent) ++ updateLabels ++
-          Map(Operator.StreamletNameLabel -> deployment.streamletName, Operator.AppIdLabel -> app.appId)
+          Map(Operator.StreamletNameLabel -> deployment.streamletName, Operator.AppIdLabel -> app.spec.appId)
+    val ownerReferences = CloudflowOwnerReferences(app)
 
     CustomResource[Spec, Status](jobSpec)
       .withMetadata(
         ObjectMeta(
           name = name,
           namespace = namespace,
           annotations = Map("prometheus.io/scrape" -> "true", "prometheus.io/port" -> PrometheusConfig.PrometheusJmxExporterPort.toString),
-          labels = labels
+          labels = labels,
+          ownerReferences = ownerReferences.list
```

Comment:
do you ever not use the list? maybe it is easier to just pass a `List[OwnerReference]` around?

Reply:
Changed it to use a `List[]` instead